ports/atmel-samd/common-hal/busio/I2C.c

```diff
@@ -71,28 +70,36 @@ void common_hal_busio_i2c_construct(busio_i2c_obj_t *self,
         mp_raise_ValueError("Invalid pins");
     }
 
-    config_i2c_master.pinmux_pad0 = sda_pinmux; // SDA
-    config_i2c_master.pinmux_pad1 = scl_pinmux; // SCL
-    config_i2c_master.buffer_timeout = 10000;
+
+    // Set up I2C clocks on sercom.
+    sercom_clock_init(sercom, sercom_index);
+
+    if (i2c_m_sync_init(&self->i2c_desc, sercom) != ERR_NONE) {
```

Comment:
The reliance of these new `_init` methods on arrays of `_configuration` structs is really annoying. Is the compiler smart enough to inline the values from the arrays or does it always take code space? I'd be tempted to init them ourselves to avoid the configuration structs and the need for the CONF values at all.

Reply:
`sercom_clock_init` is new by me (it's in `samd*1_periphals.c`). I do need the clock lookup table (maybe not the SLOW), because the numbers are random.

I think we could get rid of the `_i2cms` configuration table in `hp_sercom.c`, but it means changing the semantics of the library somewhat. I was trying to make minimal code changes to ASF4 aside from fixing bugs, so that we could merge more easily later.

The compiler can't inline the values because they're looked up by an index computed at runtime.